linux/library/src/plugin_handler.cc

```diff
@@ -0,0 +1,50 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <flutter/plugin_handler.h>
+
+#include <iostream>
+
+namespace flutter {
+
+PluginHandler::PluginHandler() {}
+
+PluginHandler::~PluginHandler() {}
+
+bool PluginHandler::AddPlugin(std::unique_ptr<Plugin> plugin) {
+  if (plugins_.find(plugin->channel()) != plugins_.end()) {
+    return false;
+  }
+  plugins_.emplace(std::make_pair(plugin->channel(), std::move(plugin)));
```

Comment:
Because emplace can call constructors, in cases like this where insert does what we want anyway using insert is clearer.

Reply:
Fixed.